index.js

```diff
@@ -18,13 +20,13 @@
 
 function paginate(query, options, callback) {
   query = query || {};
-  options = Object.assign({}, paginate.options, options);
-  let select = options.select;
+  options = options || {};
```

Comment:
paginate.options, limit=0, leanWithId=true were added in 5.0.0, tested and documented

https://github.com/edwardhotchkiss/mongoose-paginate/tree/v5.0.0#set-custom-default-options-for-all-queries
https://github.com/edwardhotchkiss/mongoose-paginate/tree/v5.0.0#zero-limit